packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -442,7 +442,13 @@ export class WorkspaceCommandContribution implements CommandContribution {
         const childUri = parent.resource.resolve(name);
         const exists = await this.fileService.exists(childUri);
         if (exists) {
-            return nls.localizeByDefault('A file or folder "{0}" already exists at this location.', this.trimFileName(name));
+            // check the nature of the child item so we can return a specific validation message
+            const existingChild = await this.fileService.resolve(childUri);
+            if (existingChild.isDirectory) {
+                return nls.localizeByDefault('A folder "{0}" already exists at this location.', this.trimFileName(name));
+            } else {
+                return nls.localizeByDefault('A file "{0}" already exists at this location.', this.trimFileName(name));
```

Comment:
As per the [i18n section in our coding guidelines](https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#internationalizationlocalization), the `nls.localizeByDefault` function should only be called when an appropriate value exists in vscode's language packs (represented by `nls.metadata.json` file). This file should not be modified, since it won't match strings that exist inside of vscode's language packs anymore.

Instead you should use `nls.localize` and introduce new keys like `theia/workspace/folderExists` and `theia/workspace/fileExists`.

Reply:
Thanks. I misunderstood that part...
Changed now to `nls.localize`.